src/oas.md

```diff
@@ -308,21 +308,101 @@ Note that some URI fields are named `url` for historical reasons, but the descri
+Within an OpenAPI Document, a Schema Object that does not have its base URI set by `$id` uses `$self` the same as any other Object, treating the OpenAPI Document as the "encapsulating entity" in accordance with RFC3986 Section 5.1.2.
+See [JSON Schema draft 2020-12 Section 8.2](https://www.ietf.org/archive/id/draft-bhutton-json-schema-01.html#section-8.2) for more information about base URIs in Schema Objects.
+
+The most common base URI source in the absence of `$self` or `$id` is the retrieval URI, in accordance with RFC3986 Section 5.1.3.
+
+Note that these rules mean that different fields in the document, particularly those in Schema Objects vs other kinds of Objects, can have different base URIs.
+Relative URI references in `$self` or `$id` are resolved against the base URI appropriate for their location in the document.
+
+Implementations SHOULD allow users to provide the intended retrieval URI along with each document, as this is necessary in situations including:
+
+* Implementations that do not support direct retrieval, which requires additional dependencies and security considerations
+* Network configurations or conditions that prevent direct retrieval
+* Test configurations that need to simulate the document being hosted in a production location
+* Documents that exist only in-memory and have no readily identifiable location (although for a single document without references to other documents, an application-specific default base URI in accordance with RFC3986 Section 5.1.4 would also be suitable)
+
+Base URIs for documents containing OpenAPI content without an OpenAPI Object at the root ignore the `$self` rule, as no `$self` can be present in such documents.
+
+##### Examples of Base URI Determination and Reference Resolution
+
+Given a retrieval URI of `https://example.com/foo/bar/openapi.yaml`, and the following OpenAPI Document:
+
+```YAML
+openapi: 3.2.0
+$self: /openapi
+info:
+  version: 1.0
+components:
+  pathItems:
+    Foo:
+      $ref: "./shared#/components/pathItems/Foo"
+```
+
+1. The base URI used to resolve `$self` is the retrieval URI, resulting in an effective `$self` value of `https://example.com/openapi` (OADs that use HTTP content negotiation to provide both JSON and YAML representations typically do not use a file extension in their self-assigned URI, to avoid needing to change URIs based on the format).
+2. The base URI for the `$ref` comes from `$self` (`https://example.com/openapi`), producing a resolved URI of `https://example.com/shared#/components/pathItems/Foo`.
+
+Relative `$self` values are often used for APIs deployed in multiple locations, such as a device management API that is hosted on each device.
+
+In the next example, the retrieval URI is irrelevant, because `$self` is already a full URI:
```

Comment:
/full URI/absolute URI/ ?

Reply:
@notEthan it is an absolute URI which is a subset of full URIs, but the important part is that it is full and therefore does not need resolving.  The absolute-ness is incidental in this particular case, and I wanted to emphasize the part that is most relevant.

Reply:
I thought I was suggesting a more precise term, but it sounds like 'full URI' is a precise term, but one I am not familiar with. I am looking but not finding quite what a full URI is or how it relates to absolute, could you point me to that information?

Reply:
The precise term is "URI", which is not very useful as most people think "URI" means "URI-reference."  There is no more precise term than that.

Reply:
And in particular, the OAS uses "URI" to mean "URI-reference" so we're left without proper terminology.

Reply:
@notEthan I have kept thinking about this.  I've long been frustrated at the lack of anything more precise than "URI" for what we want.  It plagued us with the JSON Schema spec as well, and that's where "full URI" came from.  Although at least with JSON Schema we used "URI-reference" where appropriate.  OAS further overloading "URI" makes this harder.

idk, maybe absolute URI would be better as it is also accurate and not overloaded?  Meh.

Reply:
Is that the sound of a glossary being born? :D

Reply:
@notEthan I [reworked it a bit](https://github.com/OAI/OpenAPI-Specification/pull/4389/commits/34fb6d61c1b40688fed84b90a51c36387b3c96af) to contrast "fully resolved URI" with "relative URI-reference", see if that works for you?

Reply:
I like that, I think it reads significantly more clearly.